elastic/kibana
# [Uptime] Add configurable page size to monitor list

## Summary

Related to https://github.com/elastic/kibana/issues/54031.

This change introduces the ability to specify a configurable page size to the monitor list component in the Uptime overview page.

![image](https://user-images.githubusercontent.com/18429259/77013182-eb538280-6945-11ea-9fd3-2916e5eea696.png)

Additionally, we're placing a copy of the user's selection in `localStorage` so when users set a particular browser to a particular size, the state will persist across URL changes and session expiration.

## Testing this PR

1. Set up Uptime with at least > 100 monitors.
2. Test that each selection creates the appropriate number of rows per page, and that pagination works as expected.
3. Ensure that when changing size selections with pagination data specified, the pagination data is cleared (it won't work properly with a new size).
4. Ensure that each time you change the page size, the value reflected in local storage is also updated appropriately.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
